prow/pubsub/subscriber/subscriber.go

```diff
@@ -215,18 +217,60 @@ func (prh *presubmitJobHandler) getProwJobSpec(cfg prowCfgClient, pe ProwJobEven
 		}
 	}
 	if presubmitJob == nil {
-		return nil, nil, fmt.Errorf("failed to find associated periodic job %q", pe.Name)
+		return nil, nil, fmt.Errorf("failed to find associated presubmit job %q", pe.Name)
 	}
 
 	prowJobSpec := pjutil.PresubmitSpec(*presubmitJob, *refs)
 	return &prowJobSpec, presubmitJob.Labels, nil
 }
 
 // ppostsubmitJobHandler implements jobHandler
-type postsubmitJobHandler struct{}
+type postsubmitJobHandler struct {
+	GitClient git.ClientFactory
+}
 
 func (poh *postsubmitJobHandler) getProwJobSpec(cfg prowCfgClient, pe ProwJobEvent) (*v1.ProwJobSpec, map[string]string, error) {
-	return nil, nil, errors.New("postsubmit not supported yet")
+	// postsubmit jobs require Refs to be set
+	refs := pe.Refs
+	if refs == nil {
+		return nil, nil, errors.New("refs must be supplied")
+	}
+	if len(refs.BaseSHA) == 0 {
+		return nil, nil, errors.New("baseSHA must be supplied")
+	}
+
+	var postsubmitJob *config.Postsubmit
+	org, repo, branch := refs.Org, refs.Repo, refs.BaseRef
+	orgRepo := org + "/" + repo
+	baseSHAGetter := func() (string, error) {
+		return refs.BaseSHA, nil
+	}
+
+	// This will work with inrepoconfig
+	postsubmits, err := cfg.GetPostsubmits(poh.GitClient, orgRepo, baseSHAGetter)
```

Comment:
Reminder to update this to avoid using a nil GitClient as in https://github.com/kubernetes/test-infra/pull/22876

Reply:
done